File: LeetCode/Arrays/495. Teemo Attacking/solution.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries: return 0

        poisoned = 0

        for i in range(len(timeSeries) - 1):
            poisoned += min(duration, timeSeries[i + 1] - timeSeries[i])
        
        return poisoned + duration

File: LeetCode/Arrays/495. Teemo Attacking/solution.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        if (timeSeries.length == 0) return 0;

        int poisoned = 0;

        for(int i = 0; i < timeSeries.length - 1; i++)
            poisoned += Math.min(timeSeries[i + 1] - timeSeries[i], duration);

        return poisoned + duration;
    }
}
